pkg/client/debugging_client.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+// IsClient is a global that indicates whether or not this process is a client or not.  This is used to determined
+// whether body inspection should be allowed to disallowed.  Only clients should set this value to true.  Servers
+// should not, because body inspection breaks streams
+var IsClient = false
```

Comment:
I want to call this out in particular.  It is a global that I put it here for safety to avoid intercepting streaming input and output.  Now I'm going to go hide from @smarterclayton